### ISSUE TITLE
[JENKINS-4186] Builds triggered on startup although they have not changed

### ISSUE THREAD

**Issue by timja** [timja/jenkins-gh-issues-poc-06-20#116]
<p>After upgrade from 1.318 to 1.319, Hudson decided that (probably) all of our<br/>
projects should be rebuilt because of SCM changes. However, none of them had any<br/>
changes.<br/>
This message in my "Perforce Polling Log" causes me to file the issue under the<br/>
perforce plugin - but I don't know if it's just here?</p>

<p>Workspace is offline.</p>

<p>Scheduling a new build to get a workspace.</p>

<p>Done. Took 0 ms</p>

<p>Changes found</p>

<p>Issue 4182 also concerns workspace problems in hudson 1.319, so perhaps it's a<br/>
core issue instead?</p>

---
<details><summary><i>Originally reported by <a title="torbent" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-4186" target="_blank">Builds triggered on startup although they have not changed</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2011-07-10T13:34:30+00:00
<li><b>imported</b>: 2022-06-20
</ul></i>
</details>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=alacenaire">alacenaire</a>:</i>
<p>I have the same problem with Hudson 1.319 and my SVN projects. I have :<br/>
Revision: 3465<br/>
No changes.<br/>
A SCM change trigger started this job</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>:</i>
<p>Moved to 'core' subcomponent because it appears to not be a 'perforce' problem.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>:</i>
<p>Maybe issue 4015 and issue 4145 are related to this one? They all concern<br/>
building being done on the basis of SCM changes, yet no changes have occurred.<br/>
I don't mark them as duplicates because I'm not fully sure ...</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>This is due to the internal change in Hudson 1.319 about how a workspace is managed.</p>

<p>This should happen only once after upgrade, and while I did think about if<br/>
there's any way to avoid this, I couldn't. This affects all jobs that have SCM<br/>
polling set up, regardless of SCM.</p>

<p>I'm marking this as WONTFIX, although I'm ashamed of it.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>:</i>
<p>OK, I can live with one extra build per job. I wouldn't have liked to see extra<br/>
builds every time I restart ...<br/>
Closing.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=bhalter">bhalter</a>:</i>
<p>Any chance of getting a second look at this?  </p>

<p>I'm trying to upgrade from 1.313 to the current release and stumbling over this now.  I've got about 500 builds that are managed by Hudson and having each of them need a rebuild is a particularly ugly solution for us.  Given that we need to script up work-arounds for some changes in the config.xml and to enable some new features I'm not opposed to scripting up a work around for this.  </p>

<p>Can you be more descriptive of the change to workspaces in 1.319?</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=evernat">evernat</a>:</i>
<p>Sorry bhalter but I suppose that your 500 jobs are built now.<br/>
So closing again as won't fix.</p>

**Comment by timja** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>:</i>
<p>It turned out to be an "only-once" problem. Should have been closed ages ago.</p>
